### PR TITLE
GUACAMOLE-1956: Bring Java and JavaScript dependencies up to date.

### PR DESCRIPTION
**Pull request by aleitner**

### CODE REVIEW

**Comment by mike-jumper** [comment]
@aleitner Since an issue already existed for this ([GUACAMOLE-1956](https://issues.apache.org/jira/browse/GUACAMOLE-1956)), I've closed [GUACAMOLE-1977](https://issues.apache.org/jira/browse/GUACAMOLE-1977) as a duplicate and edited the title of this PR accordingly. You'll have to amend the commit messages, though.

File: extensions/guacamole-auth-sso/modules/guacamole-auth-sso-saml/pom.xml [hunk 1]
```diff
@@ -122,20 +122,20 @@
...
-            <version>5.4.0</version>
+            <version>6.6.0</version>
         </dependency>
 
         <!-- Apache XML Security for Java (see exclusions for java-saml) -->
         <dependency>
             <groupId>org.apache.santuario</groupId>
             <artifactId>xmlsec</artifactId>
-            <version>2.2.6</version>
+            <version>4.0.2</version>
```
Comment (mike-jumper) [inline extensions/guacamole-auth-sso/modules/guacamole-auth-sso-saml/pom.xml:132]: Are these newer versions known to be binary-compatible with the versions being overridden?

File: extensions/guacamole-auth-ldap/pom.xml [hunk 2]
```diff
@@ -51,12 +51,12 @@
...
-            <version>2.1.6</version>
+            <version>2.1.7</version>
             <exclusions>
 
                 <!--
                     Replace slightly older commons-lang3 (3.13.0) with latest
-                    compatible version (3.14.0) so that we don't need two copies
+                    compatible version (3.16.0) so that we don't need two copies
                     of the same license information.
                 -->
```
Comment (mike-jumper) [inline extensions/guacamole-auth-ldap/pom.xml:61]: Is it still correct (this newer version of `api-all` is still pointing at version 3.13.3 of `commons-lang3`)?

File: doc/licenses/bouncycastle-2.0.3/README [hunk 3]
```diff
@@ -0,0 +1,7 @@
+BouncyCastle FIPS Distribution (https://www.bouncycastle.org/fips-java)
```
Comment (mike-jumper) [inline doc/licenses/bouncycastle-2.0.3/README:1]: It doesn't look like this is correct for `bcutil-fips`. The title here is identical to the title used for `bc-fips`, but the latest version of `bc-fips` is 2.0.0.

Checking https://search.maven.org/artifact/org.bouncycastle/bcutil-fips/2.0.3/jar, I see: "Bouncy Castle ASN.1 Extension and Utility APIs (FIPS Distribution)"

File: extensions/guacamole-auth-ldap/pom.xml (hunk 2 above)
Comment (aleitner) [reply extensions/guacamole-auth-ldap/pom.xml:61]: Looks like it depends on `3.14.0` and `3.15.0` still 

https://mvnrepository.com/artifact/org.apache.directory.api/api-all/2.1.7

```
Dependency convergence error for org.apache.commons:commons-lang3:3.14.0 paths to dependency are:
+-org.apache.guacamole:guacamole-auth-ldap:1.5.5
  +-org.apache.directory.api:api-all:2.1.7
    +-org.apache.commons:commons-text:1.12.0
      +-org.apache.commons:commons-lang3:3.14.0
and
+-org.apache.guacamole:guacamole-auth-ldap:1.5.5
  +-org.apache.directory.api:api-all:2.1.7
    +-org.apache.commons:commons-lang3:3.15.0
and
+-org.apache.guacamole:guacamole-auth-ldap:1.5.5
  +-org.apache.commons:commons-lang3:3.16.0
```

File: extensions/guacamole-auth-sso/modules/guacamole-auth-sso-saml/pom.xml (hunk 1 above)
Comment (aleitner) [reply extensions/guacamole-auth-sso/modules/guacamole-auth-sso-saml/pom.xml:132]: woodstox-core 6.6.0 has full compatibility with 5 https://github.com/FasterXML/woodstox/wiki/Woodstox-Release-6.6

xmlsec I couldn't find any information on compatibility between 4 and 2, but I was able to successfully sign in with saml using these updated versions

**Review by mike-jumper — APPROVED** [review]

**Pull request merged**